module/zfs/ddt.c

```diff
@@ -673,7 +687,7 @@ ddt_remove(ddt_t *ddt, ddt_entry_t *dde)
 }
 
 ddt_entry_t *
-ddt_lookup(ddt_t *ddt, const blkptr_t *bp, boolean_t add)
+ddt_lookup(ddt_t *ddt, const blkptr_t *bp, boolean_t nogrow, boolean_t *addedp)
```

Comment:
Instead of returning `*addedp`, could callers tell if a new entry is being added based on the returned dde's `ddt_phys_total_refcnt()` (if zero, it's new)?

Reply:
This does appear to work and is a nicer solution.